files/scripts/syncd.sh

```diff
@@ -150,6 +151,12 @@ stop() {
         TYPE=cold
     fi
 
+    if [[ x$sonic_asic_platform == x"mellanox" ]] && [[ x$TYPE == x"cold" ]]; then
+        debug "Stopping pmon service ahead of syncd..."
+        /bin/systemctl stop pmon
```

Comment:
do we need chipdown here?

Reply:
> do we need chipdown here?

I don't think so. But this should be double-checked with @nazariig

Reply:
@stephenxs since the pmon was removed from config reload sequence, i suggest to do pmon stop for all cases.

Reply:
@nazariig @stephenxs How long does it take to stop pmon? If we do stop pmon for all cases it affects warm shutdown. In warm shutdown every second matters

Reply:
@stepanblyschak a couple of seconds and yes, it might affect warm-boot flow.

Reply:
@nazariig @stephenxs  I must say that right now in master we are very close to the limit 3 lacp timeouts 90sec, which is going to be optimized, however a couple of seconds can kill the warm reboot flow.

Reply:
Yes. This is why pmon isn't stopped for warm-reboot. I suggest remaining the current logic.
BTW, maybe not related, is it possible that others ways, like defer the initialization of not timing-sensitive services, contributes more to optimizing warm reboot than just not to stop pmon gracefully?

Reply:
@stephenxs  This is done in this way (e.g. SNMP is delayed, however such delay seems to me more like a hack)
And right now even with such optimization sonic is very slow at startup causing trouble in warm reboot scenario, so I suggest not to add additional delays at shutdown. Also originally only syncd should stop gracefully, anything else is either killed by signal or we don't care about graceful shutdown in favor of faster reboot. Even if platform has enough time to not let LAG/BGP flapping it is better to save 2-3 sec for the case when subsequent changes in configuration may slow down the startup